ballerina/websocket_service.bal

```diff
@@ -14,54 +14,65 @@
 // specific language governing permissions and limitations
 // under the License.
 
-import ballerina/io;
 import ballerina/websocket;
-
 import graphql.parser;
+import ballerina/lang.value;
 
 isolated service class WsService {
     *websocket:Service;
 
     private final Engine engine;
     private final readonly & __Schema schema;
     private final Context context;
-    private map<[websocket:Caller, parser:OperationNode]> clientsMap;
     
     isolated function init(Engine engine, __Schema schema) {
         self.engine = engine;
         self.schema = schema.cloneReadOnly();
         self.context = new();
-        self.clientsMap = {};
-    }
-                
-    isolated remote function onClose(websocket:Caller caller, int statusCode, string reason) {
-        lock {
-            _ = self.clientsMap.remove(caller.getConnectionId());
-        }
     }
 
     isolated remote function onTextMessage(websocket:Caller caller, string data) returns websocket:Error? {
         lock {
             parser:OperationNode|ErrorDetail node = validateSubscriptionPayload(data, self.engine);
             if node is parser:OperationNode {
-                self.clientsMap[caller.getConnectionId()] = [caller, node];
-
-                stream<any, io:Error?> sourceStream = new(); // receiving stream from "getSubscriptionResponse(...)" -> Not implemented yet
-                any|io:Error? next = sourceStream.iterator().next();
-
-                while next !is io:Error? {
-                    check broadcast(self.engine, self.schema, self.context, next, self.clientsMap);
-                    next = sourceStream.iterator().next();
-                }
+                parser:Selection selection = node.getSelections()[0];
```

Comment:
I wrote it to get the field node from the operation node. Since it is a subscription query, there can be only one element in the selection nodes array. Although it can be a fragment node, a while loop is iterated until it reaches the top level field node.

Reply:
Then the Parser should validate the subscription operation to have exactly one field.

Reply:
It seems `fragment node` validation and the `duplicate field removing` should do before the `subscription operation` validation. In that case, it's better to validate using a visitor. @ThisaruGuruge WDYT?

Reply:
+1